src/internal/queue/queue.ts

```diff
@@ -189,78 +197,146 @@ export abstract class Queue {
     return Promise.all(events)
   }
 
-  protected static registerTask(
-    queueName: string,
+  protected static async registerTask(
     event: SubclassOfBaseClass,
-    slowRetryQueueOnFail?: boolean,
-    onMessage?: (job: Job) => void
+    maxConcurrentTasks: number,
+    onMessage?: (job: Job) => void,
+    signal?: AbortSignal
   ) {
-    const hasSlowRetryQueue = event.withSlowRetryQueue()
-    return Queue.getInstance().work(
-      queueName,
-      event.getWorkerOptions(),
-      async (job: Job<BasePayload>) => {
-        try {
-          if (onMessage) {
-            onMessage(job)
-          }
-          const res = await event.handle(job)
-
-          QueueJobCompleted.inc({
-            name: event.getQueueName(),
-          })
+    const queueName = event.getQueueName()
+    const deadLetterName = event.deadLetterQueueName()
+
+    const concurrentTaskCount = event.getWorkerOptions().batchSize || maxConcurrentTasks
+    try {
+      // Create dead-letter queue and the normal queue
+      const queueOptions = {
+        policy: 'standard',
+        ...event.getQueueOptions(),
+      } as const
+
+      // dead-letter
+      await this.pgBoss?.createQueue(deadLetterName, {
+        ...queueOptions,
+        name: deadLetterName,
+        retentionDays: 30,
+        retryBackoff: true,
+      })
 
-          return res
-        } catch (e) {
-          QueueJobRetryFailed.inc({
-            name: event.getQueueName(),
-          })
+      // normal queue
+      await this.pgBoss?.createQueue(queueName, {
+        name: queueName,
+        ...queueOptions,
+        deadLetter: deadLetterName,
+      })
+    } catch {
+      // no-op
+    }
+
+    return this.pollQueue(event, {
+      concurrentTaskCount,
+      onMessage,
+      signal,
+    })
+  }
 
-          try {
-            const dbJob: JobWithMetadata | null =
-              (job as JobWithMetadata).priority !== undefined
-                ? (job as JobWithMetadata)
-                : await Queue.getInstance().getJobById(job.id)
+  protected static pollQueue(
+    event: SubclassOfBaseClass,
+    queueOpts: {
+      concurrentTaskCount: number
+      onMessage?: (job: Job) => void
+      signal?: AbortSignal
+    }
+  ) {
+    const pollingInterval = event.getWorkerOptions().pollingIntervalSeconds || 5 * 1000
```

Comment:
I think this should be `(event.getWorkerOptions().pollingIntervalSeconds || 5) * 1000` 

Without the parenthesis `pollingIntervalSeconds` isn't multiplied by 1000 it's just used as-is if it's truthy